builder/builder.go

```diff
@@ -0,0 +1,143 @@
+// Package builder defines interfaces for any Docker builder to implement.
+//
+// Historically, only server-side Dockerfile interpreters existed.
+// This package allows for other implementations of Docker builders.
+package builder
+
+import (
+	"io"
+	"os"
+
+	// TODO: remove dependency on daemon
+	"github.com/docker/docker/daemon"
+	"github.com/docker/docker/image"
+	"github.com/docker/docker/runconfig"
+)
+
+// ImageID is a temporary type for an Image ID.
+// It should be replaced with an image reference.
+type ImageID string
```

Comment:
@LK4D4 I can remove the type if you prefer and just use `string` directly. I put the type to make sure we replace it later with a type from image package or a reference package (from distribution). However you want.